pentagon/migration/migrations/migration_2_6_2.py

```diff
@@ -377,19 +387,27 @@ def run(self):
 
                             hooks = cluster_spec.get("hooks")
                             if hooks:
+                                logging.debug(hooks)
                                 for hook in hooks:
                                     if hook['name'] == 'kops-hook-authenticator-config.service':
-                                        hooks.pop(hooks.index(hook))
+                                        kops_hook_index = hooks.index(hook)
+                                        logging.debug("Found kops auth hook at index %d", kops_hook_index)
+                                    else:
+                                        logging.debug("Found other existing hook %s", hook['name'])
+                                        hook['manifest'] = literal_unicode(hook['manifest'])
+
+                                logging.debug("Removing existing kops-hook-authenticator-config.service at %d", kops_hook_index)
+                                hooks.pop(kops_hook_index)
```

Comment:
Moved the pop outside the loop so that we can actually iterate over the list properly and run the literal_unicode on the other hook manifests.